paddle/fluid/framework/ir/conv_bn_fuse_pass.cc

```diff
@@ -0,0 +1,331 @@
+
+  std::unordered_set<Node*> nodes2delete;
+
+  GraphPatternDetector gpd;
+  auto* conv_input =
+      gpd.mutable_pattern()
+          ->NewNode(patterns::PDNodeName(name_scope_, "conv_input"))
+          ->AsInput()
+          ->assert_is_op_input("conv2d", "Input");
+  patterns::ConvBN conv_bn_pattern(gpd.mutable_pattern(), name_scope_);
+  conv_bn_pattern(conv_input, true /*with_eltwise_add*/);
+
+  int found_conv_bn_count = 0;
+  auto handler = [&](const GraphPatternDetector::subgraph_t& subgraph,
+                     Graph* g) {
+    VLOG(4) << "handle ConvBN fuse";
+    // OPERATORS
+    GET_IR_NODE_FROM_SUBGRAPH(conv, conv, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bias, bias, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(batch_norm, batch_norm, conv_bn_pattern);
+    // CONV inputs
+    GET_IR_NODE_FROM_SUBGRAPH(conv_weight, conv_weight, conv_bn_pattern);
+    // CONV outputs
+    GET_IR_NODE_FROM_SUBGRAPH(conv_out, conv_out, conv_bn_pattern);
+    // BIAS inputs
+    GET_IR_NODE_FROM_SUBGRAPH(bias_y, bias_y, conv_bn_pattern);
+    // BIAS outputs
+    GET_IR_NODE_FROM_SUBGRAPH(bias_out, bias_out, conv_bn_pattern);
+    // BN inputs
+    GET_IR_NODE_FROM_SUBGRAPH(bn_scale, bn_scale, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_bias, bn_bias, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_mean, bn_mean, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_variance, bn_variance, conv_bn_pattern);
+    // BN outputs
+    GET_IR_NODE_FROM_SUBGRAPH(bn_out, bn_out, conv_bn_pattern);  // Out
+    GET_IR_NODE_FROM_SUBGRAPH(bn_mean_out, bn_mean_out, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_variance_out, bn_variance_out,
+                              conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_saved_mean, bn_saved_mean, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_saved_variance, bn_saved_variance,
+                              conv_bn_pattern);
+
+    // Find bias from BN
+    auto* bn_bias_var = scope->FindVar(bn_bias->Name());
+    auto* bn_bias_tensor = bn_bias_var->GetMutable<LoDTensor>();
+
+    // Get y (bias) variable
+    auto* bias_y_tensor =
+        scope->FindVar(bias_y->Name())->GetMutable<LoDTensor>();
+
+    PADDLE_ENFORCE_EQ(bias_y_tensor->dims(), bn_bias_tensor->dims());
+
+    // Compute new weights and bias
+    auto* scale_tensor =
+        scope->FindVar(bn_scale->Name())->GetMutable<LoDTensor>();
+    auto* variance_tensor =
+        scope->FindVar(bn_variance->Name())->GetMutable<LoDTensor>();
+    auto* mean_tensor =
+        scope->FindVar(bn_mean->Name())->GetMutable<LoDTensor>();
+
+    auto std_tensor = LoDTensor();
+    std_tensor.Resize(bn_bias_tensor->dims());
+    std_tensor =
+        tensor_apply(variance_tensor, [](float x) { return x + 1e-5f; });
+
+    tensor_apply_inplace(&std_tensor, std::sqrt);
+    auto tmp_tensor =
+        tensor_apply_eltwise(*scale_tensor, std_tensor, std::divides<float>());
+    *bias_y_tensor = tensor_apply_eltwise(
```

Comment:
Consider increasing readability of the code

Reply:
as above